### PR TITLE
controller: Truncate metric history in background

### PR DESCRIPTION
**Pull request by jkosh44**
Previously, during startup the controller would sequentially truncate
the append-only metrics history and status history introspection
collections. This process would block startup and take a non-trivial
amount of time.

This commit moves the process of truncating append-only metrics history
introspection sources into the append only background write task. This
truncation no longer blocks startup and happens completely in the
background. This commit also lays the foundation to move status history
truncation into the background.

Works towards resolving MaterializeInc/database-issues#8384

### Motivation
This PR adds a known-desirable feature.

### Tips for reviewer

Reviewing one commit at a time might be easier due to code movement.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/storage-controller/src/lib.rs [hunk 1]
```diff
@@ -321,6 +317,7 @@ where
             // CollectionManager when they are created. We only bring ourselves
             // up to date with collection contents or do any preparatory
             // consolidation work when we actually start writing to them!
+            // TODO(jkosh44) This no longer works.
```
Comment (jkosh44) [inline src/storage-controller/src/lib.rs:320]: @aljoscha Is it safe to remove this entire method? From what I can tell it is being used in a couple of places.

Comment (jkosh44) [reply src/storage-controller/src/lib.rs:320]: Never mind, I see that https://github.com/MaterializeInc/materialize/pull/29699 deletes all of this code and all of it's callers. I'll probably wait for that PR to merge and then rebase this one.

File: src/storage-controller/src/lib.rs [hunk 2]
```diff
@@ -3146,8 +3160,6 @@ where
             _ => panic!("not a metrics history: {collection:?}"),
         };
 
-        let id = self.introspection_ids.lock().expect("poisoned")[&collection];
```
Comment (aljoscha) [inline src/storage-controller/src/lib.rs:3196]: We did this but had the `GlobalId` all along, must have been a vestige of an older state ... 🙈

**Review by aljoscha — APPROVED** [review]
Very nice! I don't know if the nightly failures indicate a problem with this change, but I don't think they should.

**Pull request merged**